app/src/main/java/org/mozilla/focus/browser/VideoVoiceCommandMediaSession.kt

```diff
@@ -116,11 +118,25 @@ class VideoVoiceCommandMediaSession @UiThread constructor(
 
     @OnLifecycleEvent(ON_START)
     fun onStart() {
+        isStarted = true
+
         mediaSession.isActive = true
     }
 
     @OnLifecycleEvent(ON_STOP)
     fun onStop() {
+        isStarted = false
+
+        // Videos playing when the app was backgrounded get into an inconsistent state: `paused`
+        // will return false but they won't actually be playing. This makes any user "play" commands
+        // take two clicks: one to "pause" and the second to play for real. We get around this by
+        // pausing all videos before being backgrounded. This issue doesn't affect pages with
+        // autoplay, like YouTube.
+        //
+        // The videos may send playback state update events to , which we're forced to ignore: see
+        // JavascriptVideoPlaybackStateSyncer for the code.
+        webView?.evalJS("document.querySelectorAll('video').forEach(v => v.pause());")
```

Comment:
Ugh, the webview doesn't automatically do this, even though the video stops playing? That's annoying.

This is something that I'd flag QA to look at, because it seems weird and easily broken. Please add a QA-needed flag with extra info when this gets landed.

Reply:
The WebView stops playback but the video does not consider itself paused (perhaps it's in a state similar to when it's buffering?).

QA flags sounds like a good idea.